static/src/javascripts/projects/commercial/modules/dfp/on-slot-viewable.js

```diff
@@ -5,11 +5,29 @@ import { Advert } from 'commercial/modules/dfp/Advert';
 import { getAdvertById } from 'commercial/modules/dfp/get-advert-by-id';
 import { enableLazyLoad } from 'commercial/modules/dfp/lazy-load';
 
-const viewabilityThresholdMs = 30000;
+const getViewabilityThreshold = (advert: Advert): ?number => {
+    const refreshThresholdMsDefault = 30000;
+    const refreshThresholdMsVideo = 90000;
+
+    const fluidSize = '0,0';
+
+    const sizeString = advert.size && advert.size.toString();
+    const videoSizes = ['620,1', '620,350'];
+
+    if (sizeString === fluidSize) {
+        return;
+    } else if (videoSizes.includes(sizeString)) {
+        return refreshThresholdMsVideo;
+    }
+
+    return refreshThresholdMsDefault;
+};
 
 export const onSlotViewable = (event: ImpressionViewableEvent): void => {
     const advert: ?Advert = getAdvertById(event.slot.getSlotElementId());
-    if (advert) {
+    const viewabilityThresholdMs = advert && getViewabilityThreshold(advert);
+
+    if (advert && viewabilityThresholdMs) {
```

Comment:
@LATaylor-guardian which is why I check it here - if I didn't then `flow` would complain, but it also represents the logic for working out whether we should refresh the advert; if the threshold is undefined then we shouldn't refresh.

Reply:
Ah sorry - missed the ?.